core/consensus/component.go

```diff
@@ -545,3 +568,22 @@ func newRoundTimer(round int64) (<-chan time.Time, func()) {
 	timer := time.NewTimer(roundStart + (time.Duration(round) * roundIncrease))
 	return timer.C, func() { timer.Stop() }
 }
+
+func valuesByHash(values []*anypb.Any) (map[[32]byte]*anypb.Any, error) {
+	resp := make(map[[32]byte]*anypb.Any)
```

Comment:
Should this function return error if values is empty?

Reply:
nope, empty values result in a empty map, this is required for legacy v0.13 support.